lib/upgrade_html.js

```diff
@@ -116,144 +127,15 @@ function upgradeHtml(filename) {
-          node.attribs[attrName] = '{{' + newExpression + '}}';
-        }
-        if (newDeclaration) {
-          newDeclarations.push(newDeclaration);
-        }
-      }
-    });
-    // Look for expressions in text nodes
-    var allTextNodes = [];
-    if (template) {
-      allTextNodes = findAllTextNodes(template);
-    }
-    allTextNodes.forEach(function(textNode) {
-      var matchFullExpression = textNode.data.match(/^\{\{(.+)\}\}$/);
-      var matchPartialExpression = textNode.data.match(/\{\{(.+?)\}\}/);
-      var expression, computedResult, newExpression, newDeclaration;
-      if (matchFullExpression) {
-        expression = matchFullExpression[1];
-        computedResult = upgradeJs.fixupComputedExpression(
-            'Expression' + (anonymousComputedCounter++),
-            expression);
-        newExpression = computedResult[0];
-        newDeclaration = computedResult[1];
-        if (newExpression != expression) {
-          textNode.data = '{{' + newExpression + '}}';
-        }
-        if (newDeclaration) {
-          newDeclarations.push(newDeclaration);
-        }
-      } else if (matchPartialExpression) {
-        var leadingString = textNode.data.substring(
-            0, matchPartialExpression.index);
-        expression = matchPartialExpression[1];
-        var trailingString = textNode.data.substring(
-            matchPartialExpression.index + matchPartialExpression[0].length);
-        textNode.data = leadingString;
-        var expressionWrappingElement = $('<span>');
-        $(textNode).after(expressionWrappingElement);
-        expressionWrappingElement.after(trailingString);
-
-        computedResult = upgradeJs.fixupComputedExpression(
-            'Expression' + (anonymousComputedCounter++),
-            expression);
-        newExpression = computedResult[0];
-        newDeclaration = computedResult[1];
-        expressionWrappingElement.text('{{' + newExpression + '}}');
-        if (newDeclaration) {
-          newDeclarations.push(newDeclaration);
-        }
-      } else {
-        return;
-      }
-    });
-
-    // <input value={{x}}> -> <input value={{x::input}}>
-    var inputElems = recursivelyMatchInsideTemplates(
-        $, $(template), 'input, textarea');
-    inputElems.forEach(function(inputElem) {
-      // At this point we're guaranteed that any bound expression is either a
-      // function call or a simple property binding. We don't want to match
-      // function calls here, so we exclude bindings with parens.
-      var match = inputElem.attribs.value.match(/\{\{([^\(]+)\}\}/);
-      if (!match) {
-        return;
-      }
-      inputElem.attribs.value = '{{' + match[1] + '::input}}';
-    });
+    upgradeDataBoundTemplate($, template, newDeclarations);
 
     // Upgrade the js
     $('script', polyElem).each(function(_, scriptElem) {
       // Move the script after the polymer-element.
       $(polyElem).after(scriptElem);
+      if (upgradedScriptElems.has(scriptElem)) {
```

Comment:
Oh, you're right. I was trying to prevent upgrading webcomponents.js, but that needed to happen below. Removed this check.